lib/devise_ldap_authenticatable/ldap/connection.rb

```diff
@@ -133,13 +144,19 @@ def in_group?(group_name, group_attribute = LDAP::DEFAULT_GROUP_UNIQUE_MEMBER_LI
         else
           # AD optimization - extension will recursively check sub-groups with one query
           # "(memberof:1.2.840.113556.1.4.1941:=group_name)"
-          search_result = admin_ldap.search(:base => dn,
-                            :filter => Net::LDAP::Filter.ex("memberof:1.2.840.113556.1.4.1941", group_name),
-                            :scope => Net::LDAP::SearchScope_BaseObject)
-          # Will return  the user entry if belongs to group otherwise nothing
-          if search_result.length == 1 && search_result[0].dn.eql?(dn)
-            in_group = true
+          # Search both the user's group and the Domain User groups
+          domain_user_dn = admin_ldap.search(filter: "objectSid=#{domain_user_sid}")[0][:dn]
+          [dn, domain_user_dn].each do |distinguished_name|
+            search_result = admin_ldap.search(:base => distinguished_name,
+                                              :filter => Net::LDAP::Filter.ex("memberof:1.2.840.113556.1.4.1941", group_name),
+                                              :scope => Net::LDAP::SearchScope_BaseObject)
+            # Will return  the user entry if belongs to group otherwise nothing
+            if search_result.length == 1 && search_result[0].dn.eql?(distinguished_name)
+              in_group = true
```

Comment:
i'd just return true, and outside the loop do the logging unconditionally and return false.